fdbcli/tests/metacluster_fdbcli_tests.py

```diff
@@ -42,157 +44,489 @@ def run_fdbcli_command(cluster_file, *args):
-                       "metacluster register",
-                       name,
-                       "connection_string={}".format(conn_str),
-                       'max_tenant_groups={}'.format(max_tenant_groups))
+    rc, out, err = run_fdbcli_command(
+        management_cluster_file,
+        "metacluster register",
+        name,
+        "connection_string={}".format(conn_str),
+        "max_tenant_groups={}".format(max_tenant_groups),
+    )
     if rc != 0:
         raise Exception(err)
 
 
 @enable_logging()
-def setup_metacluster(logger, management_cluster, data_clusters, max_tenant_groups_per_cluster):
+def setup_metacluster(
+    logger, management_cluster, data_clusters, max_tenant_groups_per_cluster
+):
     management_cluster_file = management_cluster[0]
     management_cluster_name = management_cluster[1]
     tenant_id_prefix = random.randint(0, 32767)
-    logger.debug('management cluster: {}'.format(management_cluster_name))
-    logger.debug('data clusters: {}'.format([name for (cf, name) in data_clusters]))
-    metacluster_create(management_cluster_file, management_cluster_name, tenant_id_prefix)
+    logger.debug("management cluster: {}".format(management_cluster_name))
+    logger.debug("data clusters: {}".format([name for (_, name) in data_clusters]))
+    metacluster_create(
+        management_cluster_file, management_cluster_name, tenant_id_prefix
+    )
+    cluster_names_to_files[management_cluster_name] = management_cluster_file
     for (cf, name) in data_clusters:
-        metacluster_register(management_cluster_file, cf, name, max_tenant_groups=max_tenant_groups_per_cluster)
+        metacluster_register(
+            management_cluster_file,
+            cf,
+            name,
+            max_tenant_groups=max_tenant_groups_per_cluster,
+        )
+        cluster_names_to_files[name] = cf
+    assert len(cluster_names_to_files) == len(data_clusters) + 1
 
 
 def metacluster_status(cluster_file):
     _, out, _ = run_fdbcli_command(cluster_file, "metacluster status")
     return out
 
 
-def setup_tenants(management_cluster_file, data_cluster_files, tenants):
-    for tenant in tenants:
-        _, output, _ = run_fdbcli_command(management_cluster_file, 'tenant create', tenant)
-        expected_output = 'The tenant `{}\' has been created'.format(tenant)
+def remove_data_cluster(management_cluster_file, data_cluster_name):
+    rc, out, err = run_fdbcli_command(
+        management_cluster_file, "metacluster remove", data_cluster_name
+    )
+    return rc, out, err
+
+
+def cleanup_after_test(management_cluster_file, data_cluster_names):
+    for data_cluster_name in data_cluster_names:
+        rc, out, err = remove_data_cluster(management_cluster_file, data_cluster_name)
+        assert 0 == rc
+    rc, out, err = run_fdbcli_command(
+        management_cluster_file, "metacluster decommission"
+    )
+    assert 0 == rc
+
+
+def create_tenant(
+    management_cluster_file,
+    data_cluster_files,
```

Comment:
This is probably similar for many of the commands, since it shouldn't be necessary for the client to specify any data cluster files for tenant management operations.

Reply:
Done